### PR TITLE
Enable ConfigMap support with "-m <namespace>:<name>"

### PR DESCRIPTION
**Pull request by glowkey**
Include default configmap as part of helm chart.
    Update example values to explain configmap usage.
    Include tests for parsing configmap.

Signed-off-by: Douglas Wightman <dwightman@nvidia.com>

### CODE REVIEW

File: pkg/dcgmexporter/parser.go [hunk 1]
```diff
@@ -127,3 +150,40 @@ func recordIsCommentOrEmpty(s []string) bool {
...
+		return nil, fmt.Errorf("Malformed configmap-data: %s", c.ConfigMapData)
+	}
+
+	var cm *corev1.ConfigMap
+	cm, err := kubeClient.CoreV1().ConfigMaps(parts[0]).Get(context.TODO(), parts[1], metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("Could not retrieve ConfigMap '%s': %v", c.ConfigMapData, err)
+	}
+
+	r := csv.NewReader(strings.NewReader(cm.Data["metrics"]))
```
Comment (nikkon-dev) [inline pkg/dcgmexporter/parser.go:166]: I'd suggest to validate that "metrics" are actually present in the map:
if metrics, ok := cm.Data["metrics"]; ok {
...
} else {
Report an error here
}

File: pkg/dcgmexporter/parser.go [hunk 2]
```diff
@@ -17,23 +17,46 @@
...
+	}
+
 	if err != nil {
-		fmt.Printf("Error: %v\n", err)
-		return nil, err
+		logrus.Infof("%v, falling back to metric file %s", err, c.CollectorsFile)
+
+		records, err = ReadCSVFile(c.CollectorsFile)
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
```
Comment (nikkon-dev) [inline pkg/dcgmexporter/parser.go:54]: Shouldn't we log this to the logrus as well instead?

File: pkg/dcgmexporter/parser.go (hunk 1 above)
Comment (glowkey) [reply pkg/dcgmexporter/parser.go:170]: I've added logic and an error if 'metrics' is not present.

File: cmd/dcgm-exporter/main.go [hunk 3]
```diff
@@ -53,6 +54,7 @@ var (
 	CLIDevices             = "devices"
 	CLINoHostname          = "no-hostname"
 	CLIUseFakeGpus         = "fake-gpus"
+	ClIConfigMapData       = "configmap-data"
```
[the remark this answers is not in the document]
Comment (glowkey) [reply cmd/dcgm-exporter/main.go]: Good catch. Fixed.

**Pull request merged**